File: Atoms/com.AndrewHazelden.UIManagerLuaExamples/Scripts/Comp/UI Manager/LineEdit.lua

local ui = fu.UIManager
local disp = bmd.UIDispatcher(ui)
local width,height = 400,200

win = disp:AddWindow({
  ID = 'MyWin',
  WindowTitle = 'My First Window',
  Geometry = {100, 100, width, height},
  Spacing = 10,
  
  ui:VGroup{
    ID = 'root',
    Margin = 50,
    
    -- Add your GUI elements here:
    ui:LineEdit{ID='MyLineTxt', Text = 'Hello Fusioneers!', PlaceholderText = 'Please Enter a few words.',},
    
    ui:Button{ID = 'PrintButton', Text = 'Print Text',},
  },
})

-- The window was closed
function win.On.MyWin.Close(ev)
    disp:ExitLoop()
end

-- Add your GUI element based event functions here:
itm = win:GetItems()

function win.On.PrintButton.Clicked(ev)
  print(itm.MyLineTxt.Text)
end

function win.On.MyLineTxt.TextChanged(ev)
  print(itm.MyLineTxt.Text)
end

win:Show()
disp:RunLoop()
win:Hide()

File: Atoms/com.AndrewHazelden.UIManagerLuaExamples/Scripts/Comp/UI Manager/LineEdit.py
ui = fu.UIManager
disp = bmd.UIDispatcher(ui)

dlg = disp.AddWindow({ "WindowTitle": "My First Window", "ID": "MyWin", "Geometry": [ 100, 100, 400, 125 ], },
	[
		ui.VGroup({ "Spacing": 10, },
		[
			# Add your GUI elements here:
			ui.LineEdit({ "ID": "MyLineTxt", "Text": "Hello Fusioneers!", "PlaceholderText": "Please Enter a few words.", "Weight": 0.5}),
			ui.Button({ "ID": "PrintButton", "Text": "Print Text", "Weight": 0.5 }),
		]),
	])

itm = dlg.GetItems()

# The window was closed
def _func(ev):
	disp.ExitLoop()
dlg.On.MyWin.Close = _func

# Add your GUI element based event functions here:

def _func(ev):
	print(itm['MyLineTxt'].Text)
dlg.On.PrintButton.Clicked = _func

def _func(ev):
	print(itm['MyLineTxt'].Text)
dlg.On.MyLineTxt.TextChanged = _func

dlg.Show()
disp.RunLoop()
dlg.Hide()

